src/PrestaShopBundle/Form/Admin/Sell/Order/UpdateOrderStatusType.php

```diff
@@ -38,12 +39,20 @@ class UpdateOrderStatusType extends AbstractType
      */
     private $orderStatusChoiceProvider;
 
+    /**
+     * @var FormChoiceAttributeProviderInterface
+     */
+    private $orderStatusChoiceAttributeProvider;
+
     /**
      * @param FormChoiceProviderInterface $orderStatusChoiceProvider
      */
-    public function __construct(FormChoiceProviderInterface $orderStatusChoiceProvider)
-    {
+    public function __construct(
```

Comment:
Nice use of the `FormChoiceAttributeProviderInterface`!!
However the form should be injected with the attributes as an array, to avoid coupling it  with the `FormChoiceProviderInterface` and `FormChoiceAttributeProviderInterface` directly Especially since in this case it is the same service that is injected..

This also allows to use the form in more various ways. The service must be used in the service definition directly. You can look at `ManufacturerAddressType` class and definition for an example

Reply:
Done